swift-sdk/Internal/IterableAPIInternal.swift

```diff
@@ -134,6 +136,18 @@ final class IterableAPIInternal: NSObject, PushTrackerProtocol, AuthProvider {
         }
     }
     
+    func setEmail(_ email: String, _ token: String? = nil) {
+        authToken = token
+        
+        self.email = email
```

Comment:
It seems that calling `self.email` will set `authToken = nil`.

Reply:
self.email will call logoutPreviousUser which will set authToken to nil.

Please add a test where the email is changed from 'user1@example.com` to `user2@example.com`. That test will fail.